safe-modules/timelock-guard.md

```diff
@@ -0,0 +1,192 @@
+4. After the specified time, anyone can then call `Safe.execTransaction` as normal and execution
+  will complete as expected.
+
+### Replayability Of Transactions
+Each scheduled transaction would be identified by a hash derived from the transaction parameters
+plus a user-defined `salt`. This would allow to uniquely identify scheduled transactions while
+allowing to schedule identical transaction calls more than once, different only in the time that 
+they will execute.
+
+### Storage Of Scheduled Transactions
+Scheduled transactions will be stored with as much data as possible, limited by projected gas block
+limits and gas required to execute the most complex expected transactions. Rich transaction data
+helps with identifying the effects of a transaction, in particular to external operators such as
+bug bounty huntesr that might lack the tools or the context from internal operators.
+
+### Transaction Cancellation Flow
+To cancel a scheduled transaction, a `cancellation_threshold` number of owners would need to signal
+that they agree with it. Cancelled transactions never will be executable, unless rescheduled with a
+different `salt` and fresh signatures.
+
+In nested safe setups, the owners of a child multisig would need to signal the rejection of the
+transaction in numbers no less than the `cancellation_threshold` of the child multisig, for the
+child multisig to signal rejection of the transaction to the parent multisig.
+
+#### Choosing a `cancellation_threshold`
+1. We define a "blocking minority" as the minimum number of `owners` that can stop the multisig
+from approving transactions, which is `min(quorum, total_owners - quorum + 1)`. If the `cancellation_threshold`
+would be less than a blocking minority then it becomes the new blocking minority as it can stop any
+transactions (including modifying ownership, or responding to liveness challenges) from execution.
+A `cancellation_threshold` that is set too low can be abused in this way by malicious owners to
+permanently block the multisig from executing transactions if a LivenessModule hasn't been
+installed. For this reason, `cancellation_threshold >= blocking_minority`.
+2. Stage 1 requires that the compromise of a 75% or less of the Security Council can't be enough
+to cause a safety-level incident. If the Security Council would be turned into a reactive multisig
+with cancellation duties instead of approval duties, a compromise of 
+`total_owners - cancellation_threshold` SC owners would be enough to remove the ability of the
+SC to cancel malicious transactions. For this reason, for the Security Council multisig,
+`cancellation_threshold =< ceiling(0.25 * total_owners)`. This is coincidentally the same as the
+`blocking_minority` for the SC multisig. Therefore, `cancellation_threshold <= blocking_minority`.
+3. A configurable `cancellation_threshold` would need to be maintained along with `total_owners`
```

Comment:
Strongly prefer no configurable `cancellation_threshold` - threshold should just be according to the above formula.